game.py

```diff
@@ -86,6 +98,23 @@ def pull_card_from_deck(self, player):
         print(f"{player.name} pulled {pulled_card} from the deck ({len(self.deck)} cards left in the deck) \n")
         player.cards.append(pulled_card)
 
+    def block_the_next_player(self):
+        self.advance_to_next_player_turn()
+        self.notify_player_was_blocked()
+
+    def block_initial_player(self):
+        self.notify_player_was_blocked()
+        self.advance_to_next_player_turn()
+
+    def notify_player_was_blocked(self):
+        blocked_player = self.players[self.player_turn]
+        print(f"{blocked_player.name}'s turn was blocked by a stop card\n")
+
+    def advance_to_next_player_turn(self):
+        self.player_turn += 1
+        if self.player_turn >= self.number_of_players:
```

Comment:
debugging tip:
after advancing self.player_turn, add a check to see if. self.player_turn is bigger than number_of_players + 1, which is an indication of some bug in the code. in this case you can either print a log to see it yourself or raise a CodeError 😉